Gems/ROS2/Code/Source/RobotImporter/SDFormat/Hooks/ROS2CameraSensorHook.cpp

```diff
@@ -0,0 +1,90 @@
+                                                                          "libgazebo_ros_openni_kinect.so" };
+        importerHook.m_supportedPluginParams = AZStd::unordered_set<AZStd::string>{};
+        importerHook.m_sdfSensorToComponentCallback = [](AZ::Entity& entity,
+                                                         const sdf::Sensor& sdfSensor) -> SensorImporterHook::ConvertSensorOutcome
+        {
+            auto* cameraSensor = sdfSensor.CameraSensor();
+            if (!cameraSensor)
+            {
+                return AZ::Failure(AZStd::string("Failed to read parsed SDFormat data of %s camera sensor", sdfSensor.Name().c_str()));
+            }
+
+            CameraSensorConfiguration cameraConfiguration;
+            cameraConfiguration.m_depthCamera = cameraSensor->HasDepthCamera();
+            cameraConfiguration.m_colorCamera = (sdfSensor.Type() != sdf::SensorType::DEPTH_CAMERA) ? true : false;
+            cameraConfiguration.m_width = cameraSensor->ImageWidth();
+            cameraConfiguration.m_height = cameraSensor->ImageHeight();
+            cameraConfiguration.m_verticalFieldOfViewDeg =
+                cameraSensor->HorizontalFov().Degree() * (cameraConfiguration.m_height / cameraConfiguration.m_width);
+            if (sdfSensor.Type() != sdf::SensorType::DEPTH_CAMERA)
+            {
+                cameraConfiguration.m_nearClipDistance = static_cast<float>(cameraSensor->NearClip());
+                cameraConfiguration.m_farClipDistance = static_cast<float>(cameraSensor->FarClip());
+            }
+            else
+            {
+                cameraConfiguration.m_nearClipDistance = static_cast<float>(cameraSensor->DepthNearClip());
+                cameraConfiguration.m_farClipDistance = static_cast<float>(cameraSensor->DepthFarClip());
+            }
+
+            SensorConfiguration sensorConfiguration;
+            sensorConfiguration.m_frequency = sdfSensor.UpdateRate();
+            if (sdfSensor.Type() != sdf::SensorType::DEPTH_CAMERA)
+            {
+                Utils::AddTopicConfiguration(
+                    sensorConfiguration, "camera_image_color", CameraConstants::ImageMessageType, CameraConstants::ColorImageConfig);
+                Utils::AddTopicConfiguration(
+                    sensorConfiguration, "color_camera_info", CameraConstants::CameraInfoMessageType, CameraConstants::ColorInfoConfig);
+            }
+            if (sdfSensor.Type() != sdf::SensorType::CAMERA)
+            {
+                Utils::AddTopicConfiguration(
+                    sensorConfiguration, "camera_image_depth", CameraConstants::ImageMessageType, CameraConstants::DepthImageConfig);
+                Utils::AddTopicConfiguration(
+                    sensorConfiguration, "depth_camera_info", CameraConstants::CameraInfoMessageType, CameraConstants::DepthInfoConfig);
+            }
+
+            if (entity.CreateComponent<ROS2FrameComponent>() &&
```

Comment:
Robot Importer uses Utils::CreateComponent from ROS2GemUtilities to create components, could you check the difference?

Reply:
The one from `ROS2GemUtilities` should be used instead. This is fixed now. It also required the addition of some setters, hence setters and getters were added.